pkg/namespacestore/reconciler.go

```diff
@@ -395,9 +400,65 @@ func (r *Reconciler) ReconcileDeletion() error {
 		}
 	}
 
+	if r.NamespaceStore.Spec.NSFS != nil {
+		pvcName := r.NamespaceStore.Spec.NSFS.PvcName
+		if !util.KubeCheck(r.DeploymentEndpoint) {
+			r.Logger.Infof("No Endpoint deployment yet. requeuing for 5 minutes")
+			return fmt.Errorf("No Endpoint Deployment. requeue again")
+		}
+		n := 0
+		if r.shouldCleanEndpointDeploymentPvc() {
+			volumes := r.DeploymentEndpoint.Spec.Template.Spec.Volumes
+			for _, volume := range volumes {
+				if volume.PersistentVolumeClaim == nil || volume.PersistentVolumeClaim.ClaimName != pvcName {
+					volumes[n] = volume
+					n++
+				}
+			}
+			volumes = volumes[:n]
+			r.DeploymentEndpoint.Spec.Template.Spec.Volumes = volumes
+		}
+		for i := range r.DeploymentEndpoint.Spec.Template.Spec.Containers {
+			container := &r.DeploymentEndpoint.Spec.Template.Spec.Containers[i]
+			if container.Name == "endpoint" {
+				n := 0
+				volumeMounts := container.VolumeMounts
+				for _, mount := range volumeMounts {
+					if mount.MountPath != "/nsfs/" + r.NamespaceStore.Name {
+						volumeMounts[n] = mount
```

Comment:
I would move "/nsfs/" to a global variable

Reply:
split into 2 separate files - irrelevant